.github/workflows/govulncheck.yml

```diff
@@ -14,6 +14,9 @@ on:
     - cron: '00 00 * * *'
   workflow_dispatch:
 
+env:
+  V2_BRANCH: ${{ inputs.ref == 'refs/heads/v2-dev' }}
```

Comment:
The `V2_BRANCH` allows to customize the execution from `main`, but also set it to true in `v2-dev` once this PR is merged into main. This additional modification will make sure all the derived branches from `v2-dev` run the right CI tests.